src/main/java/ch/njol/skript/expressions/ExprTarget.java

```diff
@@ -153,17 +157,21 @@ public String toString(@Nullable Event event, boolean debug) {
 	/**
 	 * Gets an entity's target.
 	 *
-	 * @param entity The entity to get the target of
+	 * @param entity The entity to get the target of.
 	 * @param type The exact EntityData to find. Can be null for any entity.
-	 * @return The entity's target
+	 * @return The entity's target.
 	 */
 	@Nullable
 	@SuppressWarnings("unchecked")
 	public static <T extends Entity> T getTarget(LivingEntity entity, @Nullable EntityData<T> type) {
 		if (entity instanceof Mob)
 			return ((Mob) entity).getTarget() == null || type != null && !type.isInstance(((Mob) entity).getTarget()) ? null : (T) ((Mob) entity).getTarget();
-
-		RayTraceResult result = entity.rayTraceEntities(SkriptConfig.maxTargetBlockDistance.value());
+		RayTraceResult result;
+		if (!PAPER_RAYTRACE) {
+			result = entity.getWorld().rayTraceEntities(entity.getLocation(), entity.getEyeLocation().toVector(), SkriptConfig.maxTargetBlockDistance.value(), 0.0D);
```

Comment:
Firstly, this will always return the entity itself. Like I said in my comment earlier, you need a predicate to exclude the entity itself from the raytrace. 
Secondly, why cast from the feet and not from the eyes? That seems like a recipe for user confusion.
Thirdly, .toVector() does NOT create a vector using the location's yaw/pitch, but with its XYZ components. So this isn't even pointing in the right direction. Location#getDirection() should be used instead.
Fourthly, setting raySize to 0 means all hitboxes are collapsed to 0, meaning you're really never going to hit anything. Just leave it as default.
Finally, raytraceEntities ignores blocks by default. So if you don't want to ignore blocks, you'll need to do a raytraceBlocks as well to find the maximum distance the entity cast should go.

Reply:
Spigot defaults the ray size to 0.0D

Reply:
> Spigot defaults the ray size to 0.0D

Ah right my mistake, I accidentally confirmed my wrong assumption while testing because the .toVector bug gave me a false indication that it wasn't hitting anything. 
Ignore point 4 then.

Reply:
Two and Three are copied from the Spigot methods. Maybe not so much two because it's already got the entity reference, so I didn't look what it used.

Reply:
I'll adjust 1 eventually.

Reply:
> Two and Three are copied from the Spigot methods. Maybe not so much two because it's already got the entity reference, so I didn't look what it used.

What do you mean, "three is copied from the spigot methods"? I don't see any examples of `toVector()` being used in any `raytraceEntities()` call. Regardless, using `toVector()` is just wrong in this case.